clients/drcachesim/tests/skip_unit_tests.cpp

```diff
@@ -0,0 +1,151 @@
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ * DAMAGE.
+ */
+
+/* Unit tests for the skip feature. */
+
+#include "droption.h"
+#include "zipfile_file_reader.h"
+#include "tools/view_create.h"
+
+#include <iostream>
+#include <memory>
+
+#ifndef HAS_ZIP
+#    error zipfile reading is required for this test
+#endif
+
+#define FATAL_ERROR(msg, ...)                               \
+    do {                                                    \
+        fprintf(stderr, "ERROR: " msg "\n", ##__VA_ARGS__); \
+        fflush(stderr);                                     \
+        exit(1);                                            \
+    } while (0)
+
+#define CHECK(cond, msg, ...)             \
+    do {                                  \
+        if (!(cond)) {                    \
+            fprintf(stderr, "%s\n", msg); \
+            return false;                 \
+        }                                 \
+    } while (0)
+
+static droption_t<std::string> op_trace_file(DROPTION_SCOPE_FRONTEND, "trace_file", "",
+                                             "[Required] Trace input .zip file",
+                                             "Specifies the input .zip trace file.");
+
+static droption_t<bool> op_verbose(DROPTION_SCOPE_FRONTEND, "verbose", false,
+                                   "Whether to print diagnostics",
+                                   "Whether to print diagnostics");
+
+bool
+test_skip_initial()
+{
+    int view_count = 10;
+    // Our checked-in trace has a chunk size of 20, letting us test cross-chunk
+    // skips.  We verify the chunk size below to ensure updates to that file
+    // remember to set that value.
+    // We check each skip value to ensure the view tool is outputting the
+    // expected instruction count.
+    for (int skip_instrs = 0; skip_instrs < 50; skip_instrs++) {
+        if (op_verbose.get_value())
+            std::cout << "Testing -skip_instrs " << skip_instrs << "\n";
+        // Capture cerr.
+        std::stringstream capture;
+        std::streambuf *prior = std::cerr.rdbuf(capture.rdbuf());
+        // Open the trace.
+        std::unique_ptr<reader_t> iter = std::unique_ptr<reader_t>(
+            new zipfile_file_reader_t(op_trace_file.get_value()));
+        CHECK(!!iter, "failed to open zipfile");
+        CHECK(iter->init(), "failed to initialize reader");
+        std::unique_ptr<reader_t> iter_end =
+            std::unique_ptr<reader_t>(new zipfile_file_reader_t());
+        // Run the tool.
+        std::unique_ptr<analysis_tool_t> tool =
```

Comment:
This only tests skipping from the beginning of the chunk, not from the middle.

Reply:
Added a TODO to add middle skip testing once we properly duplicate the timstamp,cpu.